File: backend/src/main/java/com/daedan/festabook/timetag/dto/TimeTagResponse.java
package com.daedan.festabook.timetag.dto;

import com.daedan.festabook.timetag.domain.TimeTag;

public record TimeTagResponse(
        Long timeTagId,
        String name
) {

    public static TimeTagResponse from(TimeTag timeTag) {
        return new TimeTagResponse(
                timeTag.getId(),
                timeTag.getName()
        );
    }
}

File: android/app/src/main/java/com/daedan/festabook/data/model/response/place/TimeTagResponse.kt
package com.daedan.festabook.data.model.response.place

import com.daedan.festabook.domain.model.TimeTag
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class TimeTagResponse(
    @SerialName("name")
    val name: String,
    @SerialName("timeTagId")
    val timeTagId: Long,
)

fun TimeTagResponse.toDomain() =
    TimeTag(
        name = name,
        timeTagId = timeTagId,
    )
